quinn-udp/src/unix.rs

```diff
@@ -116,22 +97,28 @@ fn init(io: &std::net::UdpSocket) -> io::Result<()> {
 
     let addr = io.local_addr()?;
 
-    // macos and ios do not support IP_RECVTOS on dual-stack sockets :(
-    if addr.is_ipv4() || ((!cfg!(any(target_os = "macos", target_os = "ios"))) && !only_v6(io)?) {
-        let on: libc::c_int = 1;
-        let rc = unsafe {
-            libc::setsockopt(
-                io.as_raw_fd(),
-                libc::IPPROTO_IP,
-                libc::IP_RECVTOS,
-                &on as *const _ as _,
-                mem::size_of_val(&on) as _,
-            )
-        };
-        if rc == -1 {
-            return Err(io::Error::last_os_error());
+    // mac and ios do not support IP_RECVTOS on dual-stack sockets :(
+    // older macos versions also don't have the flag and will error out if we don't ignore it
+    let on: libc::c_int = 1;
+    let rc = unsafe {
+        libc::setsockopt(
```

Comment:
Because `IP_RECTOS` is an IPv4-specific option, we should only be running this when `only_v6(io)` is false, not unconditionally.

Reply:
ah my bad! i'll restore that check